addons/stock/report/product_templates.xml

```diff
@@ -42,24 +41,24 @@
         <template id="label_lot_template_view">
             <t t-foreach="docs" t-as="lot">
                 <t t-translation="off">
-^XA
+^XA^CI28
 ^FO100,50
-^A0N,44,33^FD<t t-out="lot.product_id.display_name"/>^FS
+^A0N,44,33^FD<t t-out="lot['display_name_markup']"/>^FS
 ^FO100,100
-^A0N,44,33^FDLN/SN: <t t-out="lot.name"/>^FS
+^A0N,44,33^FDLN/SN: <t t-out="lot['name']"/>^FS
                     <t t-if="env.user.has_group('stock.group_stock_lot_print_gs1')">
-                        <t t-if="lot.product_id.valid_ean" t-set="final_barcode" t-value="'01' + '0' * (14 - len(lot.product_id.barcode)) + lot.product_id.barcode"/>

```

Comment:
@clesgow My fix has changed a bit because of this new part. I modified the test so that it should also cover this part.